### PR TITLE
Separated some UI components into `@optuna/react`

### PR DESCRIPTION
**Pull request by porink0424**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

NA

## What does this implement/fix? Explain your changes.

- Separated some UI components into `@optuna/react` as the first step to make codes common
  - In this PR, the following components were moved into `@optuna/react` as common codes:
    - `DataGrid`
    - `PlotHistory`
    - `TrialTable`
    - `PlotlyDarkMode`

### CODE REVIEW

File: biome.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
 		],
 		"ignore": [
 			"optuna_dashboard/ts/components/PlotlyColorTemplates.ts",
+			"tslib/react/src/components/PlotlyDarkMode.ts",
 			"standalone_app/src/PlotlyDarkMode.ts",
```
Comment (c-bata) [inline biome.json:18]: Can we just remove this line?

File: standalone_app/package.json [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
+		"@optuna/react": "../tslib/react",
+		"@optuna/storage": "../tslib/storage",
+		"module-workers-polyfill": "^0.3.2",
+		"notistack": "^3.0.1",
+		"optuna": "../rustlib/pkg",
+		"plotly.js-dist-min": "^2.30.0",
+		"react": "^18.2.0",
+		"react-dom": "^18.2.0",
+		"react-router-dom": "^6.22.3"
+	}
```
Comment (c-bata) [inline standalone_app/package.json:37]: [question] Should we use a tab for the indentation?

File: tslib/react/src/utils/loadStorageFromFile.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { JournalFileStorage, SQLite3Storage } from "@optuna/storage"
```
Comment (c-bata) [inline tslib/react/src/utils/loadStorageFromFile.ts:1]: Could you move this file into `tslib/storage` or just leaving it in the `standalone_app`?

**Review by c-bata — COMMENTED** [review]
Changes looks almost good to me. I left some suggestions though.

File: standalone_app/package.json [hunk 4]
```diff
@@ -1,37 +1,41 @@
...
+		"notistack": "^3.0.1",
+		"optuna": "../rustlib/pkg",
+		"plotly.js-dist-min": "^2.30.0",
+		"react": "^18.2.0",
+		"react-dom": "^18.2.0",
+		"react-router-dom": "^6.22.3"
+	},
+	"optionalDependencies": {
+		"@rollup/rollup-linux-x64-gnu": "^4.13.2"
+	}
```
Comment (porink0424) [inline standalone_app/package.json:40]: Memo:These lines are added to resolve the error due to rollup dependency (module not found error: `Error: Cannot find module @rollup/rollup-linux-x64-gnu`)

ref: https://stackoverflow.com/questions/77569907/error-in-react-vite-project-due-to-rollup-dependency-module-not-found

File: biome.json (hunk 1 above)
Comment (porink0424) [reply biome.json:18]: Removed L18 (`"standalone_app/src/PlotlyDarkMode.ts"`), not L17 (`"tslib/react/src/components/PlotlyDarkMode.ts"`), thank you for your catch!

File: standalone_app/package.json (hunk 2 above)
Comment (porink0424) [reply standalone_app/package.json]: No specific reason for my change here, so I replaced tab with spaces.
Thanks for your comment.

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**